src/python/deepgnn/graph_engine/snark/synchronized.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Synchronized server - client setup."""
+from deepgnn.graph_engine.snark.distributed import Server, Client as DistributedClient
+import ray
```

Comment:
This creates an explicit dependency on ray for deepgnn-ge package. We want to avoid it and keep dependencies minimal. Please, make sure it is an [extra instead](https://peps.python.org/pep-0508/#extras).

Reply:
As long as the user does not import synchronized.py, they are not required to have Ray - e.g. all the unit tests work even though ray is not specified as a dependency.

Reply:
We no longer have backends, we just use Ray data directly on servers.

Reply:
Sure, but I need to see how it is done. So far, backend code is still [present in this branch](https://github.com/microsoft/DeepGNN/tree/cdieckhaus/synchronize/src/python/deepgnn/graph_engine/backends).

Reply:
This is only for TF support, it is fully removed from pytorch at this point. What do you want me to update the code to do?

Reply:
In the current form it is more confusing to have multiple synchronized modules. So I would rather not create this one at all, because it is not used in any example and documentation is contrived.

Reply:
Close this PR?

Reply:
Yes. Synchronized module can be updated after tf is migrated.